amundsen_application/static/js/components/SearchPage/index.tsx

```diff
@@ -42,57 +42,67 @@ import {
   isLoading: boolean;
   tables: TableSearchResults;
   dashboards: DashboardSearchResults;
   users: UserSearchResults;
 }
 
 export interface DispatchFromProps {
-  searchAll: (term: string, options?: SearchAllOptions) => SearchAllRequest;
+  searchAll: (term: string, selectedTab: ResourceType, pageIndex: number) => SearchAllRequest;
   searchResource: (resource: ResourceType, term: string, pageIndex: number) => SearchResourceRequest;
+  updateSearchTab: (selectedTab: ResourceType) => UpdateSearchTabRequest;
 }
 
 export type SearchPageProps = StateFromProps & DispatchFromProps & RouteComponentProps<any>;
 
-interface SearchPageState {
-  selectedTab: ResourceType;
-}
-
-export class SearchPage extends React.Component<SearchPageProps, SearchPageState> {
+export class SearchPage extends React.Component<SearchPageProps> {
   public static defaultProps: Partial<SearchPageProps> = {};
 
   constructor(props) {
     super(props);
-
-    this.state = {
-      selectedTab: ResourceType.table,
-    };
   }
 
   componentDidMount() {
-    const params = qs.parse(this.props.location.search);
-    const { searchTerm, pageIndex, selectedTab } = params;
-    const { term, index, currentTab } = this.getSanitizedUrlParams(searchTerm, pageIndex, selectedTab);
-    this.setState({ selectedTab: currentTab });
-    if (term !== '') {
-      this.props.searchAll(term, this.createSearchOptions(index, currentTab));
-      if (currentTab !== selectedTab || pageIndex !== index) {
-        this.updatePageUrl(term, currentTab, index);
+    const urlParams = this.getUrlSearchParams(this.props.location.search);
+    const propParams = this.getParamsFromGlobalState();
+
+    if (urlParams.term === '') {
+      if (propParams.term !== '') {
+        this.updatePageUrl(propParams.term, propParams.tab, propParams.index, true);
+      }
+    } else {
+      if (urlParams.term !== propParams.term) {
+        this.props.searchAll(urlParams.term, urlParams.tab, urlParams.index);
+        this.updatePageUrl(urlParams.term, urlParams.tab, urlParams.index, true);
       }
     }
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.props.location.search !== prevProps.location.search) {
-      const params = qs.parse(this.props.location.search);
-      const { searchTerm, pageIndex, selectedTab } = params;
-      const { term, index, currentTab } = this.getSanitizedUrlParams(searchTerm, pageIndex, selectedTab);
-      this.setState({ selectedTab: currentTab });
-      const prevTerm = prevProps.searchTerm;
-      if (term !== prevTerm) {
-        this.props.searchAll(term, this.createSearchOptions(index, currentTab));
-      }
+  componentDidUpdate(prevProps: SearchPageProps) {
+    const currParams = this.getUrlSearchParams(this.props.location.search);
+    const prevParams = this.getUrlSearchParams(prevProps.location.search);
+    const propParams = this.getParamsFromGlobalState();
+
+    // ULR params and global state is in sync, no further action needed
```

Comment:
To the same flavor as my comment above in componentDidMount, creating methods that tell us whether or not we should update or what specific case we're in will help. Well named functions will also remove the need to explaining everything in comments -- though there's likely still no harm in documenting the intended functionality here so we can remember exactly what this logic is months from now.